atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/transaction/impl/AdvisoryLockConditionSuppliers.java

```diff
@@ -0,0 +1,105 @@
+
+import org.apache.commons.lang3.Validate;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.common.base.Supplier;
+import com.google.common.collect.ImmutableSet;
+import com.palantir.atlasdb.transaction.api.LockAcquisitionException;
+import com.palantir.lock.HeldLocksToken;
+import com.palantir.lock.LockClient;
+import com.palantir.lock.LockRequest;
+import com.palantir.lock.LockService;
+
+public final class AdvisoryLockConditionSuppliers {
+
+    private static final Logger log = LoggerFactory.getLogger(AdvisoryLockConditionSuppliers.class);
+
+    private static final int NUM_RETRIES = 10;
+
+    private AdvisoryLockConditionSuppliers() {}
+
+    public static Supplier<AdvisoryLocksCondition> get(LockService lockService, Iterable<HeldLocksToken> lockTokens,
+            Supplier<LockRequest> lockSupplier) {
+        return () -> {
+            Set<HeldLocksToken> externalLocks = ImmutableSet.copyOf(lockTokens);
+            ExternalLocksCondition externalCondition = externalLocks.isEmpty()
+                    ? null
+                    : new ExternalLocksCondition(lockService, externalLocks);
+
+            LockRequest lockRequest = lockSupplier.get();
+            if (lockRequest != null) {
+                Validate.isTrue(lockRequest.getVersionId() == null, "Using a version id is not allowed");
+                HeldLocksToken newToken = acquireLock(lockService, lockRequest);
+                TransactionLocksCondition transactionCondition = new TransactionLocksCondition(lockService, newToken);
+
+                return externalCondition == null
+                        ? transactionCondition
+                        : new CombinedLocksCondition(externalCondition, transactionCondition);
+            }
+
+            return externalCondition == null
+                    ? NO_LOCKS_CONDITION
+                    : externalCondition;
+        };
+    }
+
+    private static HeldLocksToken acquireLock(LockService lockService, LockRequest lockRequest) {
+        int failureCount = 0;
+        while (true) {
+            HeldLocksToken response;
+            try {
+                response = lockService.lockAndGetHeldLocks(LockClient.ANONYMOUS.getClientId(), lockRequest);
+            } catch (InterruptedException e) {
+                Thread.currentThread().interrupt();
+                throw new RuntimeException(e);
+            }
+            if (response == null) {
+                RuntimeException ex = new LockAcquisitionException(
+                        "Failed to lock using the provided lock request: " + lockRequest);
+                log.warn("Count not lock successfully", ex);
+                ++failureCount;
+                if (failureCount >= NUM_RETRIES) {
+                    log.warn("Failing after {} tries", failureCount, ex);
+                    throw ex;
+                }
+                AbstractTransactionManager.sleepForBackoff(failureCount);
+                continue;
```

Comment:
nit: Doesn't seem like there's anything after this statement, so this probably isn't needed